### PR TITLE
[Merged by Bors] - chore: adaptations for nightly-2024-06-17

### PR DESCRIPTION
**Pull request by kim-em**

### CODE REVIEW

File: Mathlib/Data/Nat/Defs.lean [hunk 1]
```diff
@@ -151,20 +151,10 @@ lemma le_of_pred_lt : ∀ {m}, pred m < n → m ≤ n
 
 lemma lt_iff_add_one_le : m < n ↔ m + 1 ≤ n := by rw [succ_le_iff]
 #align nat.lt_iff_add_one_le Nat.lt_iff_add_one_le
-
--- Just a restatement of `Nat.lt_succ_iff` using `+1`.
-lemma lt_add_one_iff : m < n + 1 ↔ m ≤ n := Nat.lt_succ_iff
-#align nat.lt_add_one_iff Nat.lt_add_one_iff
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Nat/Defs.lean:157]: Restore this align please

File: Mathlib/Data/Nat/Defs.lean [hunk 2]
```diff
@@ -151,20 +151,10 @@ lemma le_of_pred_lt : ∀ {m}, pred m < n → m ≤ n
...
-
 -- A flipped version of `lt_add_one_iff`.
 lemma lt_one_add_iff : m < 1 + n ↔ m ≤ n := by simp only [Nat.add_comm, Nat.lt_succ_iff]
 #align nat.lt_one_add_iff Nat.lt_one_add_iff
 
--- This is true reflexively, by the definition of `≤` on ℕ,
--- but it's still useful to have, to convince Lean to change the syntactic type.
-lemma add_one_le_iff : m + 1 ≤ n ↔ m < n := Iff.rfl
-#align nat.add_one_le_iff Nat.add_one_le_iff
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Nat/Defs.lean:166]: And this

File: Mathlib/Data/Nat/Defs.lean [hunk 3]
```diff
@@ -433,9 +423,6 @@ protected lemma le_of_mul_le_mul_right (h : a * c ≤ b * c) (hc : 0 < c) : a 
 protected alias mul_sub := Nat.mul_sub_left_distrib
 protected alias sub_mul := Nat.mul_sub_right_distrib
 
-protected lemma mul_sub_one (a b : ℕ) : a * (b - 1) = a * b - a := by rw [Nat.mul_sub, Nat.mul_one]
-protected lemma sub_one_mul (a b : ℕ) : (a - 1) * b = a * b - b := by rw [Nat.sub_mul, Nat.one_mul]
-
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Nat/Defs.lean:438]: Why are these removed?

Comment (semorrison) [reply Mathlib/Data/Nat/Defs.lean:438]: Because they were upstreamed.

File: Mathlib/Data/Nat/Defs.lean [hunk 4]
```diff
@@ -151,20 +151,10 @@ lemma le_of_pred_lt : ∀ {m}, pred m < n → m ≤ n
 
 lemma lt_iff_add_one_le : m < n ↔ m + 1 ≤ n := by rw [succ_le_iff]
 #align nat.lt_iff_add_one_le Nat.lt_iff_add_one_le
```
Comment (semorrison) [inline Mathlib/Data/Nat/Defs.lean:153]:
```suggestion
#align nat.lt_iff_add_one_le Nat.lt_iff_add_one_le

#align nat.lt_add_one_iff Nat.lt_add_one_iff
```

File: Mathlib/Data/Nat/Defs.lean [hunk 5]
```diff
@@ -151,20 +151,10 @@ lemma le_of_pred_lt : ∀ {m}, pred m < n → m ≤ n
...
-
--- Just a restatement of `Nat.lt_succ_iff` using `+1`.
-lemma lt_add_one_iff : m < n + 1 ↔ m ≤ n := Nat.lt_succ_iff
-#align nat.lt_add_one_iff Nat.lt_add_one_iff
-
 -- A flipped version of `lt_add_one_iff`.
 lemma lt_one_add_iff : m < 1 + n ↔ m ≤ n := by simp only [Nat.add_comm, Nat.lt_succ_iff]
 #align nat.lt_one_add_iff Nat.lt_one_add_iff
```
Comment (semorrison) [inline Mathlib/Data/Nat/Defs.lean:157]:
```suggestion

#align nat.add_one_le_iff Nat.add_one_le_iff
```

File: Mathlib/Data/Nat/Defs.lean (hunk 2 above)
Comment (semorrison) [reply Mathlib/Data/Nat/Defs.lean:166]: I am really looking forward to these being gone!

**Review by TwoFX — APPROVED** [review]
Thanks!
bors r+
